pydicom/filereader.py

```diff
@@ -230,6 +233,15 @@ def data_element_generator(fp,
     element_struct_unpack = element_struct.unpack
     defer_size = size_in_bytes(defer_size)
 
+    tag_set = None
+    if specific_tags is not None:
+        tag_set = set()
```

Comment:
I'm not sure. I should sit down with pen and paper to double check. Remain me which is the case where checking not none differs from implicitly check inside the set. Or its more readable than using something like : 
```py
if tag not in tag_set.union([ a, b, c])  and tag_set is not empty:
```